### PR TITLE
preflight: support setting up custom repo for ubuntu

### PR DESCRIPTION
**Pull request by ycy1766**
Add custom repository configuration for ubuntu.

### CODE REVIEW

**Review by guits — CHANGES_REQUESTED** [review]
please, add a commit message and a sign-off to your commit

thanks

**Comment by ycy1766** [comment]
@guits  Sorry, I added a commit message and a sign-off

File: cephadm-preflight.yml [hunk 1]
```diff
@@ -213,6 +213,15 @@
                 filename: ceph
                 update_cache: no
 
+            - name: configure Ceph custom repositories
+              when: ceph_origin == 'custom'
+              apt_repository:
+                repo: "deb {{ item.baseurl }}/ {{ ansible_facts['distribution_release'] }} {{ item.components }}"
```
Comment (guits) [inline cephadm-preflight.yml:220]: `item.components` will be a mandatory parameter for debian-based OS only, right?

Comment (ycy1766) [reply cephadm-preflight.yml:220]: Yes , As you know `item.components ` variable is a variable used only in debian-based OS.
When this is merged I will update doc for debian-based variable ( `item.components `) .

Comment (guits) [reply cephadm-preflight.yml:220]: > When this is merged I will update doc for debian-based variable ( `item.components `) .

why don't you update the documentation in this PR?

Comment (ycy1766) [reply cephadm-preflight.yml:220]: I found out that commits to doc are done separately for this project. 
I added this variable in doc. 
What else should I add or change?

Comment (guits) [reply cephadm-preflight.yml:220]: > I found out that commits to doc are done separately for this project.

That's true I'm personally used to sending separate commits for the documentation but this project has no such requirements.

File: doc/source/index.rst [hunk 2]
```diff
@@ -253,6 +253,11 @@ priority
 **description**
   The priority of this repository.
 
+components
+########
```
Comment (guits) [inline doc/source/index.rst]:
```suggestion
components
##########
```

File: doc/source/index.rst [hunk 3]
```diff
@@ -253,6 +253,11 @@ priority
...
+**description**
+  The components of this ubuntu/debian repository.
```
Comment (guits) [inline doc/source/index.rst:260]:
```suggestion
  This is a Debian OS-based family parameter only.
  The components of this ubuntu/debian repository.
```

**Review by guits — CHANGES_REQUESTED** [review]

**Comment by guits** [comment]
@ycy1766 can you squash your commits? thanks!

**Comment by ycy1766** [comment]
I will close this PR and edit it to re-open PR

**Pull request closed without merging**